pkg/resource/training_job/custom_set_output.go

```diff
@@ -17,75 +17,50 @@
 package training_job
 
 import (
-	"context"
-
-	ackv1alpha1 "github.com/aws-controllers-k8s/runtime/apis/core/v1alpha1"
-	svcapitypes "github.com/aws-controllers-k8s/sagemaker-controller/apis/v1alpha1"
+	svccommon "github.com/aws-controllers-k8s/sagemaker-controller/pkg/common"
 	"github.com/aws/aws-sdk-go/aws"
 	svcsdk "github.com/aws/aws-sdk-go/service/sagemaker"
-	corev1 "k8s.io/api/core/v1"
 )
 
-// customDescribeTrainingJobSetOutput sets the resource ResourceSynced condition to False if
-// TrainingJob is being modified by AWS. It has an additional check on the debugger status.
-func (rm *resourceManager) customDescribeTrainingJobSetOutput(
-	ctx context.Context,
-	r *resource,
-	resp *svcsdk.DescribeTrainingJobOutput,
-	ko *svcapitypes.TrainingJob,
-) (*svcapitypes.TrainingJob, error) {
-	trainingJobStatus := resp.TrainingJobStatus
-	debuggerRuleInProgress := false
-	if resp.DebugRuleEvaluationStatuses != nil {
-		for _, rule := range resp.DebugRuleEvaluationStatuses {
-			if rule.RuleEvaluationStatus != nil && *rule.RuleEvaluationStatus == svcsdk.RuleEvaluationStatusInProgress {
-				debuggerRuleInProgress = true
-				rm.customSetOutput(r, aws.String(svcsdk.TrainingJobStatusInProgress), ko)
-				break
-			}
-		}
+var (
+	trainingJobModifyingStatuses = []string{
+		svcsdk.TrainingJobStatusInProgress,
+		svcsdk.TrainingJobStatusStopping,
 	}
-
-	if !debuggerRuleInProgress {
-		rm.customSetOutput(r, trainingJobStatus, ko)
+	ruleModifyingStatuses = []string{
+		svcsdk.RuleEvaluationStatusInProgress,
+		svcsdk.RuleEvaluationStatusStopping,
 	}
+	resourceName = resourceGK.Kind
+)
 
-	return ko, nil
-}
-
-// customSetOutput sets ConditionTypeResourceSynced condition to True or False
-// based on the trainingJobStatus on AWS so the reconciler can determine if a
-// requeue is needed
-func (rm *resourceManager) customSetOutput(
-	r *resource,
-	trainingJobStatus *string,
-	ko *svcapitypes.TrainingJob,
-) {
-	if trainingJobStatus == nil {
+// customDescribeTrainingJobSetOutput sets the resource ResourceSynced condition to False if
+// TrainingJob is being modified by AWS. It has an additional check on the debugger and profiler status.
```

Comment:
nit: "It has an additional check", makes it sound like there is an additional check occurring on the debugger and profiler status that is different from checking to see if it is being modified by AWS.

Reply:
reworded